test/lib/pgBackRestTest/Common/RunTest.pm

```diff
@@ -341,6 +344,59 @@ sub testResult
             $bDone = true;
         }
     } while (!$bDone);
+
+    # If we get here then test any warning message
+    if (defined($strWarnMessage))
+    {
+        testWarning($fnSub, $strWarnMessage, $bTestExactWarnMessage);
+    }
+}
+
+####################################################################################################################################
+# testWarning
+####################################################################################################################################
+sub testWarning
+{
+    my $fnSub = shift;
+    my $strWarnMessage = shift;
+    my $bTestExact = shift;
+
+    if (ref($fnSub) ne 'CODE')
+    {
+        confess "a function must be called to test a warning message";
+    }
+
+    # Save the current log levels and set the file level to warn and timestam false
+    my ($strLogLevelFile, $strLogLevelConsole, $strLogLevelStdErr, $bLogTimestamp) = logLevel();
+    logLevelSet(WARN, undef, undef, false);
+
+    # Clear the cache just in case
+    logFileCacheClear();
+
+    # Run the function
+    $fnSub->();
+
+    # Restore the log levels
+    logLevelSet($strLogLevelFile, $strLogLevelConsole, $strLogLevelStdErr, $bLogTimestamp);
```

Comment:
This needs to be in an exception block so log levels are restored even on exception.